### PR TITLE
Expand code-like blocks in nested contexts

### PR DESCRIPTION
**Pull request by ettersi**
Fixes https://github.com/JuliaDocs/Documenter.jl/issues/491. This PR is heavily inspired by the previous work in https://github.com/JuliaDocs/Documenter.jl/pull/1637 and https://github.com/JuliaDocs/Documenter.jl/pull/1747

_Edit by @mortenpi: close #1747 ._

### CODE REVIEW

**Comment by fredrikekre** [comment]
Can you add a note in the changelog?

**Comment by ettersi** [comment]
Done!

**Comment by ettersi** [comment]
Not sure why some tests fail, but I suspect it's caused by something outside the scope of this PR. The tests used to pass before the last two commits, and since then I've only changed the changelog.

File: src/Utilities/Selectors.jl [hunk 1]
```diff
@@ -163,7 +163,7 @@ Selectors.dispatch(MySelector, args...)
 """
 function dispatch(::Type{T}, x...) where T <: AbstractSelector
     types = get!(selector_subtypes, T) do
-        sort(subtypes(T); by = order)
+        sort(leave_subtypes(T); by = order)
```
Comment (mortenpi) [inline src/Utilities/Selectors.jl:166]:
```suggestion
        sort(leaf_subtypes(T); by = order)
```

File: src/Utilities/Selectors.jl [hunk 2]
```diff
@@ -174,6 +174,28 @@ function dispatch(::Type{T}, x...) where T <: AbstractSelector
...
 end
 
+"""
+Return a list of all subtypes of `T` which do not have further subtypes.
+
+The returned list includes subtypes of subtypes, and it does not distinguish
+between concrete types (i.e. types which are guaranteed not to have subtypes)
+and abstract types (which may or may not have subtypes).
+"""
+function leave_subtypes(::Type{T}) where T
```
Comment (mortenpi) [inline src/Utilities/Selectors.jl:184]:
```suggestion
function leaf_subtypes(::Type{T}) where T
```

**Review by mortenpi — COMMENTED** [review]
This looks great! Just a couple of tiny nitpicks.

Could you add the new example page to `pages` here: https://github.com/JuliaDocs/Documenter.jl/blob/998d26e568a20e3b1764d0c2cef30ebe4d2563ae/test/examples/make.jl#L181-L213 ?

The test failures indeed look unrelated.

**Comment by mortenpi** [comment]
I can replicate the test failures locally and outside of Julia as well, and it looks like it's this Ubuntu bug: https://bugs.launchpad.net/ubuntu/+source/git/+bug/1993586

**Comment by ettersi** [comment]
Done

**Comment by mortenpi** [comment]
Thanks a lot @ettersi!

**Comment by ettersi** [comment]
Thank you! Documenter is a great tool, and I'm very grateful to everyone who has helped shape it into what it is today!

**Comment by ettersi** [comment]
Btw, is there a rough guesstimate for when the next version will be released?

**Comment by mortenpi** [comment]
> Btw, is there a rough guesstimate for when the next version will be released?

Not really, no. I did some thinking, and I would like to get [a few more things into 0.28](https://github.com/JuliaDocs/Documenter.jl/milestone/107). People get somewhat annoyed with breaking Documenter releases, so I would like batch up as many (breaking or potentially breaking) things as we can.

However, a good thing about Documenter deployments is that you can just commit a `Manifest.toml` and run with `Documenter#master` for your docs. There is no particular need to depend on released versions.

**Pull request merged**